shared/lib/multichain/chain-agnostic-permission-utils/caip-accounts.ts

```diff
@@ -85,3 +92,47 @@ export function isInternalAccountInPermittedAccountIds(
     });
   });
 }
+
+/**
+ * Checks if an internal account is connected to any of the permitted accounts
+ * based on scope matching
+ *
+ * @param internalAccount - The internal account to check against permitted accounts
+ * @param permittedAccounts - Array of CAIP account IDs that are permitted
+ * @returns True if the account is connected to any permitted account
+ */
+export function isInternalAccountInPermittedAccountIds(
+  internalAccount: InternalAccount,
+  permittedAccounts: CaipAccountId[],
+): boolean {
+  const parsedInteralAccountScopes = internalAccount.scopes.map((scope) => {
+    return parseScopeString(scope);
+  });
+
+  return isAddressWithParsedScopesInPermittedAccountIds(
+    internalAccount.address,
+    parsedInteralAccountScopes,
+    permittedAccounts,
+  );
+}
+
+/**
+ * Checks if an CAIP account ID is connected to any of the permitted accounts
+ * based on scope matching
+ *
+ * @param accountId - The CAIP account ID to check against permitted accounts
+ * @param permittedAccounts - Array of CAIP account IDs that are permitted
+ * @returns True if the account is connected to any permitted account
+ */
+export function isAccountIdInPermittedAccountIds(
```

Comment:
I think the key thing to highlight about this function is that the "permittedAccounts" will contain 0 as references, which is why we can't just do a simple `includes()` check?

Reply:
right